backend/app/repositories/task.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Optional
+from app.repositories.base import BaseRepository, get_repository
+from app.settings import settings
+from app.models.task import TaskResult
+from app.repositories.base import NotFoundError, OSNotFoundError
+from app.worker.app import celery_app
+
+
+class TaskRepository(BaseRepository[TaskResult]):
+    model_class = TaskResult
+    index = settings.CELERY_INDEX
+    id_prefix = "celery-task-meta-"
+
+    def get(self, id: str) -> TaskResult:
```

Comment:
I understand that you implemented a result backend for OpenSearch, but I'm not convinced you need to go through `AsyncResult` in this context: you have access here to your server and are able to make direct queries.

I would say that it's better to make a query to OpenSearch to get the result by task ID. This way, you don't need to pollute the repository with Celery concepts.

Reply:
Yes, I originally went this way but found out that the was a good bit of logic baked into `AsyncResult` which I liked not having to implement. I'll take a closer look to see what would be involved.

Reply:
Updated to query OpenSearch instead of using AsyncResult.